aya/src/util.rs

```diff
@@ -200,9 +201,25 @@ pub fn online_cpus() -> Result<Vec<u32>, io::Error> {
 
 /// Get the number of possible cpus.
 ///
+/// Caches the value into thread local value on the first successful access
 /// See `/sys/devices/system/cpu/possible`.
 pub fn nr_cpus() -> Result<usize, io::Error> {
-    Ok(possible_cpus()?.len())
+    // per https://www.kernel.org/doc/Documentation/core-api/cpu_hotplug.rst
```

Comment:
Why a thread local rather than a global static? Why not https://doc.rust-lang.org/stable/std/sync/struct.OnceLock.html?

Reply:
It doesn't have overhead on synchronization at all. Initialization is pretty cheap, so calling it `num thread` times is not a problem. Also, `OnceLock` has an MSRV of 1.70 which is incompatible with aya's MSRV. And I'm not sure how to cache fallible operations given that `io::Error` is not clonable:)

Reply:
Probably want https://doc.rust-lang.org/stable/std/cell/struct.OnceCell.html#method.get_or_try_init which is still unstable.